### PR TITLE
Isc

### LINKED ISSUE #65: isc calendar export
Let a user export the events to an .isc file.

### PR DESCRIPTION
**Pull request by alanhdu**
image

Add export button. Clicking it downloads an `.isc` file that can be read by Google calendar.

Closes #65

### CODE REVIEW

File: cal/isc.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from datetime import datetime
+
+from icalendar import Calendar as iCalendar, Event as iEvent, vText, vDatetime
+
+
+def to_icalendar(events):
+    """ Converts events to an text (.isc format)
```
Comment (davidhao3300) [inline cal/isc.py]: events to an text => events to text

File: cal/static/jsx/query.jsx [hunk 2]
```diff
@@ -58,10 +58,27 @@ var RFiltering = React.createClass({
...
 })
 
+var RDownload = React.createClass({
+    onClick: function() {
+        var uids = this.props.userList.map(function (user) {
+            return user.id
+        });
+        url = "/isc/?" + $.param({event_ids: uids});
+        // kind of a hack, don't know how else to do it
+        window.location.href = url;
```
Comment (davidhao3300) [inline cal/static/jsx/query.jsx:69]: I don't think there's a much better alternative than this.

**Comment by davidhao3300** [comment]
LGTM, you may want to ask Dan to look over the Javascript syntax.

**Comment by alanhdu** [comment]
@danrschlosser Could you look over the javascript here?

File: cal/__init__.py [hunk 3]
```diff
@@ -98,3 +100,12 @@ def search(searchfield):
                                 .filter(Event.start < week_later)
 
     return jsonify(data=[event.to_json() for event in search_results])
+
+
+@app.route("/isc/", methods=["GET"])
+def to_isc():
+    event_ids = request.args.getlist("event_ids[]", type=int)
```
Comment (schlosser) [inline cal/__init__.py:108]: let's add a comment explaining what's going on here

**Comment by alanhdu** [comment]
Merging.

**Comment by alanhdu** [comment]
Merging.

File: cal/static/jsx/query.jsx (hunk 2 above)
Comment (schlosser) [inline cal/static/jsx/query.jsx:69]: Check this out.  Are users going to spend any amount of time on the page we're redirecting to?  If that's not a problem, then this way is the accepted way.

http://stackoverflow.com/questions/200337/whats-the-best-way-to-automatically-redirect-someone-to-another-webpage

File: cal/static/jsx/query.jsx [hunk 4]
```diff
@@ -58,10 +58,27 @@ var RFiltering = React.createClass({
...
+    },
+    render: function() {
+        return (
+            <button onClick={ this.onClick.bind(this) }> Export Events </button>
```
Comment (schlosser) [inline cal/static/jsx/query.jsx:73]: why do we need to do `.bind(this)` here?

File: cal/static/jsx/query.jsx (hunk 2 above)
Comment (schlosser) [inline cal/static/jsx/query.jsx:69]: So I think we're good here, you can remove the comment

**Pull request merged**